homeassistant/components/seventeentrack/sensor.py

```diff
@@ -210,51 +202,66 @@ def state(self):
     @property
     def unique_id(self):
         """Return a unique, HASS-friendly identifier for this entity."""
-        return ENTITY_ID_TEMPLATE.format(
+        return UNIQUE_ID_TEMPLATE.format(
             self._data.account_id, self._tracking_number)
 
     async def async_update(self):
         """Update the sensor."""
         await self._data.async_update()
 
-        if not self._data.packages:
+        if not self.available:
+            await self._remove()
```

Comment:
We don't need to await this operation. Use `hass.async_create_task` and pass your callback to it.

Reply:
yes, well spot